brave/brave-core
# Added Shields settings section

Resolves https://github.com/brave/brave-browser/issues/12782

- Added **Shield settings** to `brave://settings/content`:
  - Shield status (`up` / `down` or `enabled` / `disabled`), navigates to `/shields` settings subpage
  - Removed `Upgrade connections to HTTPS` option
  - ~Upgrade connections to HTTPS (`true` / `false`), navigates to `/shields` settings subpage~ (removed)
- ~Replaced `brave.https_everywhere_default` by `profile.default_content_setting_values.httpUpgradableResouces`~  (removed)
- Each subpage allows to include/exclude domains same way as for other settings

https://user-images.githubusercontent.com/2965009/187741643-42513b26-921e-4cee-bb84-38e8502fdec6.mp4

**Updated screenshots**

<img width="797" alt="image" src="https://user-images.githubusercontent.com/2965009/195711135-bb3f7db9-ff1a-4385-bf6f-7b718a73b8b0.png">

<img width="701" alt="image" src="https://user-images.githubusercontent.com/2965009/195711254-553b6d9a-3da7-41c3-84da-e49ec1d4caf8.png">


## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Enable/Disable shields on different domains, make sure all domains are visible on `brave://settings/content/shields`
- Try to add/remove domains using `Add` buttons for each sections and check if shields meets this configuration
